shadows/framework/src/main/java/org/robolectric/shadows/ShadowParcelFileDescriptor.java

```diff
@@ -46,15 +49,32 @@ protected void __constructor__(ParcelFileDescriptor wrapped) {
   }
 
   @Implementation
-  protected static ParcelFileDescriptor open(File file, int mode) throws FileNotFoundException {
-    ParcelFileDescriptor pfd;
+  protected static ParcelFileDescriptor adoptFd(int fd) {
     try {
-      Constructor<ParcelFileDescriptor> constructor =
-          ParcelFileDescriptor.class.getDeclaredConstructor(FileDescriptor.class);
-      pfd = constructor.newInstance(new FileDescriptor());
+      File file = new File(Files.readSymbolicLink(Paths.get("/proc/self/fd/" + fd)).toString());
```

Comment:
It breaks Java portability， for `/proc/self/fd/` is Linux only.

Reply:
What about getting file with passed fd, and changing its permission? Does this fd created by `ShadowParcelFileDescriptor`?

Reply:
Yeah, I think it's a good idea, and we can add a constraint on the fd.